share/wake/lib/system/runner.wake

```diff
@@ -148,17 +148,34 @@ export def fuseRunner: Runner =
     | makeJSONRunner
 
 export def defaultRunner: Runner =
-    require Some _ = getenv "WAKE_LOCAL_JOB_CACHE"
-    else fuseRunner
-
-    # The fuseRunner does not actully mount over / and instead uses the
-    # the host root as the sandbox root. This means that wakeroot will
-    # change from depending on where wake is being run from. As a hack
-    # to work around this we abuse the fact that the fuseRunner only
-    # works in relative paths to make different runs consistent. Ideally
-    # you'd have a more complex sandbox that kept the wakeroot at a
-    # consistent place across runs.
-    mkJobCacheRunner (\_ Pass "") "/workspace" fuseRunner
+    require Some config = getenv "WAKE_REMOTE_CACHE"
+    else
+        require Some _ = getenv "WAKE_LOCAL_JOB_CACHE"
+        else fuseRunner
+
+        # The fuseRunner does not actully mount over / and instead uses the
+        # the host root as the sandbox root. This means that wakeroot will
+        # change from depending on where wake is being run from. As a hack
+        # to work around this we abuse the fact that the fuseRunner only
+        # works in relative paths to make different runs consistent. Ideally
+        # you'd have a more complex sandbox that kept the wakeroot at a
+        # consistent place across runs.
+        mkJobCacheRunner (\_ Pass "") "/workspace" fuseRunner
```

Comment:
Nit, but sometimes these nested requires are easier to read as a match.

Reply:
This is another example where `when` would be _so much_ nicer

```
def defaultRunner =
    when Some config = getenv "WAKE_REMOTE_CACHE"
        mkRscRunner ...
    when Some _ = getenv "WAKE_LOCAL_JOB_CACHE"
        mkJobCacheRunner ...
    fuseRunner
```

I'm still partial to the decreased nesting of `require` over `match` here though